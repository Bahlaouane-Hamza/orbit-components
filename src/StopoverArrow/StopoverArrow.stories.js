// @flow
import * as React from "react";
import { storiesOf, setAddon } from "@storybook/react";
import styles from "@sambego/storybook-styles";
import chaptersAddon from "react-storybook-addon-chapters";
import { select, text, withKnobs } from "@storybook/addon-knobs/react";

import STOPS from "./consts";

import StopoverArrow from "./";

setAddon(chaptersAddon);

storiesOf("StopoverArrow", module)
  .addDecorator(withKnobs)
  .addDecorator(
    styles({
      padding: "20px",
    }),
  )
  .addWithChapters("Playground", () => {
    const stops = select("stops", Object.values(STOPS), STOPS.ZERO);
    const dataTest = text("dataTest", "test");

    return {
      info:
        "You can try all possible configurations of this component. However, check Orbit.Kiwi for more detailed design guidelines.",
      chapters: [
        {
          sections: [
            {
              sectionFn: () => <StopoverArrow stops={stops} dataTest={dataTest} />,
            },
          ],
        },
      ],
    };
  });
