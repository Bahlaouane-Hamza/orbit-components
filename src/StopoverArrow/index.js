// @flow
import * as React from "react";
import styled from "styled-components";

import defaultTokens from "../defaultTokens";
import STOPS from "./consts";

import type { Props } from "./index";

const StyledArrow = styled.svg`
  vertical-align: middle;
  fill: currentColor;
  color: ${({ theme }) => theme.orbit.paletteInkLight}; // TODO: create token colorStopoverArrow
  width: 50px; // TODO: Create token widthStopoverArrow
  height: 7px; // TODO: Create token heightStopoverArrow
`;

StyledArrow.defaultProps = {
  theme: defaultTokens,
};

const Stops = ({ stops }: Props) => {
  if (stops === STOPS.ONE) {
    return (
      <path d="M50.0353 3.53564l-3.5357 3.53541-.707-.70704L48.121 4.0354H26.9498c-.2317 1.14087-1.2407 1.99976-2.4499 1.99976-1.2093 0-2.2183-.85889-2.45-1.99976H-.00012234l-8e-8-1H22.0499c.2317-1.14111 1.2405-2.00024 2.45-2.00024 1.2095 0 2.2182.85913 2.4499 2.00024H48.121L45.7926.707276 46.4999 3e-7 50.0353 3.53564zm-27.0354-.00048c0 .82837.6716 1.5 1.5 1.5.8283 0 1.5-.67163 1.5-1.5s-.6717-1.5-1.5-1.5c-.8284 0-1.5.67163-1.5 1.5z" />
    );
  } else if (stops === STOPS.TWO) {
    return (
      <path d="M50.0352 3.53564l-.7071.70704-2.8283 2.82836-.7073-.70703 2.3286-2.32861H30.95c-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.2092 0-2.2183-.85889-2.45-1.99976h-4.1c.0185-.09082.0319-.18359.0402-.27759-.1132 1.27564-1.185 2.27735-2.4902 2.27735-1.2092 0-2.2183-.85889-2.45-1.99976H0v-1h17.0498c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h17.1704L45.7925.707275 46.4995 0l2.8286 2.82837.7071.70727zM18 3.53516c0 .82837.6716 1.5 1.5 1.5s1.5-.67163 1.5-1.5-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5zm10.5 1.5c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5.6716 1.5 1.5 1.5z" />
    );
  } else if (stops === STOPS.THREE) {
    return (
      <path d="M46.4995 0l2.8286 2.82837.7071.70727-.7071.70704-2.8283 2.82836-.7073-.70703 2.3286-2.32861H35.95c-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.2092 0-2.2183-.85889-2.45-1.99976h-4.1c.0161-.07886.0285-.15918.0368-.24072-.0005.00464-.001.00879-.0014.01343-.0083.0769-.0203.15258-.0354.22729-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.2092 0-2.2183-.85889-2.45-1.99976h-4.1c-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.3765 0-2.4934-1.11304-2.5-2.48804.0005.08105.0046.16113.0125.24023.0085.08399.0209.16675.0375.24805H0v-1h13.0498c-.0176.08569-.0305.17358-.0386.26245.1197-1.26929 1.1883-2.26269 2.4888-2.26269 1.3806 0 2.5 1.11938 2.5 2.5 0-.09229-.0049-.18335-.0146-.27295-.0083-.07666-.0201-.15235-.0352-.22681h4.0996c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h12.1704L45.7925.707275 46.4995 0zM15.5 5.03516c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5.6716 1.5 1.5 1.5zm7.5-1.5c0 .82837.6716 1.5 1.5 1.5s1.5-.67163 1.5-1.5-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5zm10.5 1.5c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5.6716 1.5 1.5 1.5z" />
    );
  } else if (stops === STOPS.FOUR) {
    return (
      <path d="M36 3.53516c0 .82842.6716 1.5 1.5 1.5s1.5-.67158 1.5-1.5c0-.82843-.6716-1.5-1.5-1.5s-1.5.67157-1.5 1.5zm-.9965494.13248906C35.0011597 3.62377597 35 3.57960342 35 3.53516c0-1.38072 1.1193-2.5 2.5-2.5 1.3294839 0 2.416601 1.03778231 2.4954234 2.3474725-.0030675-.05274555-.007836-.10502861-.0142234-.1568025-.0081-.06421-.0183-.12769-.031-.19043h8.1709L45.7927.707275 46.5 0l3.5354 3.53564-3.5356 3.5354-.7073-.70703 2.3286-2.32861H39.95a2.4679841 2.4679841 0 0 0 .0008383-.00415845C39.720673 5.1743025 38.7108592 6.03516 37.5 6.03516c-1.21083 0-2.2206243-.86081602-2.4507698-2.00385522L35.05 4.0354h-4.1c.0017-.00854.0034-.01733.0051-.02588C30.7329 5.1626 29.718 6.03516 28.5 6.03516c-1.3188 0-2.3994-1.02149-2.4934-2.31641.0078.10767.0225.21338.0434.31665h-4.1c.0112-.05493.0205-.1106.028-.16675.0054-.03955.0098-.07934.0132-.11963.0056-.06714.0088-.13525.0088-.20385-.0007.16821-.0178.33154-.05.49023-.2322 1.14038-1.241 1.99976-2.45 1.99976-1.2087 0-2.2175-.85938-2.45-1.99976l-.007-.03589c.0022.01196.0046.02393.007.03589h-4.1c-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.20923 0-2.21826-.85889-2.44995-1.99976H0v-1h8.0498l-.00219.01123C8.2749 1.89966 9.28638 1.03516 10.5 1.03516c1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c-.0034.01733-.0066.03467-.0098.052.211-1.16699 1.2322-2.05224 2.46-2.05224 1.3245 0 2.4087 1.031 2.4946 2.33374-.0014-.02369-.0034-.04737-.0056-.0708-.0085-.08912-.0215-.17676-.0388-.2627h4.0996c-.0327.16138-.0498.32861-.0498.49976 0 .04443.00115.0886175.0034375.13250125zm.0457796.36365572a2.4892309 2.4892309 0 0 1-.0118508-.06304804zm4.9015562-.00008263c.0168281-.08359295.0294864-.16869521.0377486-.25508048zM10.5 5.03516c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5c-.82837 0-1.5.67163-1.5 1.5s.67163 1.5 1.5 1.5zm7.5-1.5c0 .82837.6716 1.5 1.5 1.5s1.5-.67163 1.5-1.5-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5zm10.5 1.5c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5.6716 1.5 1.5 1.5z" />
    );
  }
  return (
    <path d="M49.293 4.2L50 3.5l-.706-.7L46.467 0l-.707.7 2.327 2.305H0v.99h48.087L45.76 6.3l.707.7z" />
  );
};

const StopoverArrow = ({ stops = STOPS.ZERO, dataTest }: Props) => (
  <StyledArrow preserveAspectRatio="xMidYMid meet" viewBox="0 0 50 7" data-test={dataTest}>
    <Stops stops={stops} />
  </StyledArrow>
);

export default StopoverArrow;
