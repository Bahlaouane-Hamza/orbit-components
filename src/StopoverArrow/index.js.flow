// @flow
import type { Globals } from "../common/common.js.flow";

export type Props = {|
  +stops: "0" | "1" | "2" | "3" | "4",
  ...Globals,
|};

declare export default React$ComponentType<Props>;
