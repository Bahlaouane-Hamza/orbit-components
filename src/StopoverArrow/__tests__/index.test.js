// @flow
import * as React from "react";
import { mount } from "enzyme";

import StopoverArrow from "../index";
import STOPS from "../consts";
import defaultTokens from "../../defaultTokens";

describe("StopoverArrow", () => {
  const stops = STOPS.FOUR;
  const dataTest = "test";
  const component = mount(<StopoverArrow stops={stops} dataTest={dataTest} />);

  it("should have passed props", () => {
    expect(component.find("Stops").prop("stops")).toBe(stops);
    expect(component.render().prop("preserveAspectRatio")).toBe("xMidYMid meet");
    expect(component.render().prop("viewBox")).toBe("0 0 50 7");
    expect(component.render().prop("data-test")).toBe(dataTest);
  });
  it("should have styles", () => {
    expect(component).toHaveStyleRule("color", defaultTokens.orbit.paletteInkLight);
  });
  it("should match snapshot", () => {
    expect(component).toMatchSnapshot();
  });
});
