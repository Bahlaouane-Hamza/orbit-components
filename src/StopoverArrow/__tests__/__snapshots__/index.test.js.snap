// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StopoverArrow should match snapshot 1`] = `
.c0 {
  vertical-align: middle;
  fill: currentColor;
  color: #7f91a8;
  width: 50px;
  height: 7px;
}

<StopoverArrow
  dataTest="test"
  stops="4"
>
  <StopoverArrow__StyledArrow
    data-test="test"
    preserveAspectRatio="xMidYMid meet"
    theme={
      Object {
        "orbit": Object {
          "backgroundAlertCritical": "#fae8e8",
          "backgroundAlertInfo": "#e0f6ff",
          "backgroundAlertSuccess": "#e7f3e8",
          "backgroundAlertWarning": "#fcf1cd",
          "backgroundBadgeCritical": "#fae8e8",
          "backgroundBadgeDark": "#46515e",
          "backgroundBadgeInfo": "#e0f6ff",
          "backgroundBadgeNeutral": "#f5f7f9",
          "backgroundBadgeSuccess": "#e7f3e8",
          "backgroundBadgeWarning": "#fcf1cd",
          "backgroundBadgeWhite": "#fff",
          "backgroundBody": "#f5f7f9",
          "backgroundButtonBordered": "transparent",
          "backgroundButtonBorderedActive": "#fff",
          "backgroundButtonBorderedHover": "#f5f7f9",
          "backgroundButtonCritical": "#d21c1c",
          "backgroundButtonCriticalActive": "#b21717",
          "backgroundButtonCriticalHover": "#bd1919",
          "backgroundButtonFacebook": "#3b5998",
          "backgroundButtonFacebookActive": "#354f88",
          "backgroundButtonFacebookHover": "#385490",
          "backgroundButtonGoogle": "#f5f7f9",
          "backgroundButtonGoogleActive": "#d6dee6",
          "backgroundButtonGoogleHover": "#e5eaef",
          "backgroundButtonInfo": "#0176D2",
          "backgroundButtonInfoActive": "#0064b2",
          "backgroundButtonInfoHover": "#006abd",
          "backgroundButtonLinkPrimary": "transparent",
          "backgroundButtonLinkPrimaryActive": "#d6dee6",
          "backgroundButtonLinkPrimaryHover": "#e5eaef",
          "backgroundButtonLinkSecondary": "transparent",
          "backgroundButtonLinkSecondaryActive": "#d6dee6",
          "backgroundButtonLinkSecondaryHover": "#e5eaef",
          "backgroundButtonPrimary": "#00a991",
          "backgroundButtonPrimaryActive": "#008f7b",
          "backgroundButtonPrimaryHover": "#009882",
          "backgroundButtonSecondary": "#e8edf1",
          "backgroundButtonSecondaryActive": "#cad5df",
          "backgroundButtonSecondaryHover": "#d9e1e8",
          "backgroundButtonSuccess": "#46B655",
          "backgroundButtonSuccessActive": "#3fa34c",
          "backgroundButtonSuccessHover": "#42ac50",
          "backgroundButtonWarning": "#f9971e",
          "backgroundButtonWarningActive": "#e68206",
          "backgroundButtonWarningHover": "#f48a06",
          "backgroundCard": "#fff",
          "backgroundCarrierLogo": "transparent",
          "backgroundCountryFlag": "transparent",
          "backgroundIllustration": "transparent",
          "backgroundInput": "#fff",
          "backgroundInputDisabled": "#e8edf1",
          "backgroundModal": "#fff",
          "backgroundSeparator": "#e8edf1",
          "backgroundServiceLogo": "transparent",
          "borderColorButtonCriticalBordered": "#d21c1c",
          "borderColorButtonCriticalBorderedActive": "#b21717",
          "borderColorButtonCriticalBorderedHover": "#bd1919",
          "borderColorButtonFacebookBordered": "#3b5998",
          "borderColorButtonFacebookBorderedActive": "#354f88",
          "borderColorButtonFacebookBorderedHover": "#385490",
          "borderColorButtonGoogleBordered": "#46515e",
          "borderColorButtonGoogleBorderedActive": "#38404b",
          "borderColorButtonGoogleBorderedHover": "#3f4854",
          "borderColorButtonInfoBordered": "#0176D2",
          "borderColorButtonInfoBorderedActive": "#0064b2",
          "borderColorButtonInfoBorderedHover": "#006abd",
          "borderColorButtonPrimaryBordered": "#00a991",
          "borderColorButtonPrimaryBorderedActive": "#008f7b",
          "borderColorButtonPrimaryBorderedHover": "#009882",
          "borderColorButtonSecondaryBordered": "#46515e",
          "borderColorButtonSecondaryBorderedActive": "#38404b",
          "borderColorButtonSecondaryBorderedHover": "#3f4854",
          "borderColorButtonSuccessBordered": "#46B655",
          "borderColorButtonSuccessBorderedActive": "#3fa34c",
          "borderColorButtonSuccessBorderedHover": "#42ac50",
          "borderColorButtonWarningBordered": "#f9971e",
          "borderColorButtonWarningBorderedActive": "#e68206",
          "borderColorButtonWarningBorderedHover": "#f48a06",
          "borderColorCard": "#e8edf1",
          "borderColorCheckboxRadio": "#bac7d5",
          "borderColorCheckboxRadioActive": "#46515e",
          "borderColorCheckboxRadioError": "#d21c1c",
          "borderColorCheckboxRadioFocus": "#0176D2",
          "borderColorCheckboxRadioHover": "#7f91a8",
          "borderColorInput": "#bac7d5",
          "borderColorInputActive": "#94a8be",
          "borderColorInputError": "#d21c1c",
          "borderColorInputErrorFocus": "#d21c1c",
          "borderColorInputErrorHover": "#bd1919",
          "borderColorInputFocus": "#0176D2",
          "borderColorInputHover": "#a6b6c8",
          "borderColorModal": "#e8edf1",
          "borderColorTableCell": "#bac7d5",
          "borderRadiusBadge": "12px",
          "borderRadiusCircle": "50%",
          "borderRadiusLarge": "6px",
          "borderRadiusNormal": "3px",
          "borderRadiusSmall": "2px",
          "borderStyleCard": "solid",
          "borderStyleInput": "solid",
          "borderWidthCard": "1px",
          "borderWidthInput": "1px",
          "borderWidthInputFocus": "2px",
          "boxShadowActionable": "0 2px 4px 0 rgba(23, 27, 30, 0.1)",
          "boxShadowButtonFocus": "0 0 4px 1px rgba(1, 118, 210, 0.3)",
          "boxShadowElevatedLevel1": "0 4px 12px 0 rgba(23, 27, 30, 0.3)",
          "boxShadowModal": "0 20px 60px 0 rgba(23, 27, 30, 0.4)",
          "colorAlertIconCritical": "#d21c1c",
          "colorAlertIconInfo": "#0176D2",
          "colorAlertIconSuccess": "#46B655",
          "colorAlertIconWarning": "#f9971e",
          "colorFormLabel": "#46515e",
          "colorFormLabelFilled": "#7f91a8",
          "colorHeading": "#171b1e",
          "colorHeadingInverted": "#fff",
          "colorIconAttention": "#171b1e",
          "colorIconCheckboxRadio": "#00a991",
          "colorIconCheckboxRadioDisabled": "#bac7d5",
          "colorIconCritical": "#d21c1c",
          "colorIconInfo": "#0176D2",
          "colorIconInput": "#bac7d5",
          "colorIconPrimary": "#46515e",
          "colorIconRadioButton": "#00a991",
          "colorIconRadioButtonDisabled": "#bac7d5",
          "colorIconSecondary": "#7f91a8",
          "colorIconSuccess": "#46B655",
          "colorIconTertiary": "#bac7d5",
          "colorIconWarning": "#f9971e",
          "colorInfoCheckBoxRadio": "#7f91a8",
          "colorPlaceholderInput": "#bac7d5",
          "colorPlaceholderInputError": "#d21c1c",
          "colorPlaceholderInputFile": "#7f91a8",
          "colorPlaceholderInputFileError": "#d21c1c",
          "colorTextAlertCritical": "#650808",
          "colorTextAlertInfo": "#003A69",
          "colorTextAlertLink": "#171b1e",
          "colorTextAlertSuccess": "#065d12",
          "colorTextAlertWarning": "#a93610",
          "colorTextAttention": "#171b1e",
          "colorTextBadgeCritical": "#d21c1c",
          "colorTextBadgeDark": "#fff",
          "colorTextBadgeInfo": "#0176D2",
          "colorTextBadgeNeutral": "#171b1e",
          "colorTextBadgeSuccess": "#46B655",
          "colorTextBadgeWarning": "#f9971e",
          "colorTextBadgeWhite": "#171b1e",
          "colorTextButtonCritical": "#fff",
          "colorTextButtonCriticalActive": "#fff",
          "colorTextButtonCriticalBordered": "#d21c1c",
          "colorTextButtonCriticalBorderedActive": "#b21717",
          "colorTextButtonCriticalBorderedHover": "#bd1919",
          "colorTextButtonCriticalHover": "#fff",
          "colorTextButtonFacebook": "#fff",
          "colorTextButtonFacebookActive": "#fff",
          "colorTextButtonFacebookBordered": "#3b5998",
          "colorTextButtonFacebookBorderedActive": "#354f88",
          "colorTextButtonFacebookBorderedHover": "#385490",
          "colorTextButtonFacebookHover": "#fff",
          "colorTextButtonFilled": "#fff",
          "colorTextButtonFilledActive": "#fff",
          "colorTextButtonFilledHover": "#fff",
          "colorTextButtonGoogle": "#46515e",
          "colorTextButtonGoogleActive": "#38404b",
          "colorTextButtonGoogleBordered": "#46515e",
          "colorTextButtonGoogleBorderedActive": "#38404b",
          "colorTextButtonGoogleBorderedHover": "#3f4854",
          "colorTextButtonGoogleHover": "#3f4854",
          "colorTextButtonInfo": "#fff",
          "colorTextButtonInfoActive": "#fff",
          "colorTextButtonInfoBordered": "#0176D2",
          "colorTextButtonInfoBorderedActive": "#0064b2",
          "colorTextButtonInfoBorderedHover": "#006abd",
          "colorTextButtonInfoHover": "#fff",
          "colorTextButtonLinkPrimary": "#00a991",
          "colorTextButtonLinkPrimaryActive": "#008f7b",
          "colorTextButtonLinkPrimaryHover": "#009882",
          "colorTextButtonLinkSecondary": "#46515e",
          "colorTextButtonLinkSecondaryActive": "#38404b",
          "colorTextButtonLinkSecondaryHover": "#3f4854",
          "colorTextButtonPrimary": "#fff",
          "colorTextButtonPrimaryActive": "#fff",
          "colorTextButtonPrimaryBordered": "#00a991",
          "colorTextButtonPrimaryBorderedActive": "#008f7b",
          "colorTextButtonPrimaryBorderedHover": "#009882",
          "colorTextButtonPrimaryHover": "#fff",
          "colorTextButtonSecondary": "#46515e",
          "colorTextButtonSecondaryActive": "#38404b",
          "colorTextButtonSecondaryBordered": "#46515e",
          "colorTextButtonSecondaryBorderedActive": "#38404b",
          "colorTextButtonSecondaryBorderedHover": "#3f4854",
          "colorTextButtonSecondaryHover": "#3f4854",
          "colorTextButtonSuccess": "#fff",
          "colorTextButtonSuccessActive": "#fff",
          "colorTextButtonSuccessBordered": "#46B655",
          "colorTextButtonSuccessBorderedActive": "#3fa34c",
          "colorTextButtonSuccessBorderedHover": "#42ac50",
          "colorTextButtonSuccessHover": "#fff",
          "colorTextButtonWarning": "#fff",
          "colorTextButtonWarningActive": "#fff",
          "colorTextButtonWarningBordered": "#f9971e",
          "colorTextButtonWarningBorderedActive": "#e68206",
          "colorTextButtonWarningBorderedHover": "#f48a06",
          "colorTextButtonWarningHover": "#fff",
          "colorTextCritical": "#d21c1c",
          "colorTextError": "#d21c1c",
          "colorTextInfo": "#0176D2",
          "colorTextInput": "#46515e",
          "colorTextInputDisabled": "#bac7d5",
          "colorTextInputPrefix": "#7f91a8",
          "colorTextLinkPrimary": "#00a991",
          "colorTextLinkPrimaryHover": "#00a991",
          "colorTextLinkSecondary": "#171b1e",
          "colorTextLinkSecondaryHover": "#00a991",
          "colorTextPrimary": "#46515e",
          "colorTextSecondary": "#7f91a8",
          "colorTextSuccess": "#46B655",
          "colorTextWarning": "#f9971e",
          "colorTextWhite": "#fff",
          "durationFast": "0.15s",
          "durationNormal": "0.3s",
          "durationSlow": "0.4s",
          "fontFamily": "\\"Roboto\\", -apple-system, \\".SFNSText-Regular\\", \\"San Francisco\\", \\"Segoe UI\\", \\"Helvetica Neue\\", \\"Lucida Grande\\", sans-serif",
          "fontSizeButtonLarge": "16px",
          "fontSizeButtonNormal": "14px",
          "fontSizeButtonSmall": "12px",
          "fontSizeFormFeedback": "12px",
          "fontSizeFormLabel": "14px",
          "fontSizeHeadingDisplay": "40px",
          "fontSizeHeadingTitle1": "28px",
          "fontSizeHeadingTitle2": "22px",
          "fontSizeHeadingTitle3": "16px",
          "fontSizeInputNormal": "14px",
          "fontSizeInputSmall": "14px",
          "fontSizeMessage": "14px",
          "fontSizeTextLarge": "16px",
          "fontSizeTextNormal": "14px",
          "fontSizeTextSmall": "12px",
          "fontWeightBold": "700",
          "fontWeightHeadingDisplay": "700",
          "fontWeightHeadingTitle1": "700",
          "fontWeightHeadingTitle2": "500",
          "fontWeightHeadingTitle3": "500",
          "fontWeightLinks": "500",
          "fontWeightMedium": "500",
          "fontWeightNormal": "400",
          "heightBadge": "24px",
          "heightButtonLarge": "52px",
          "heightButtonNormal": "44px",
          "heightButtonSmall": "32px",
          "heightCheckbox": "20px",
          "heightCountryFlag": "auto",
          "heightIconLarge": "32px",
          "heightIconMedium": "24px",
          "heightIconSmall": "16px",
          "heightIllustrationMedium": "200px",
          "heightIllustrationSmall": "90px",
          "heightInputGroupSeparatorNormal": "24px",
          "heightInputGroupSeparatorSmall": "16px",
          "heightInputLarge": "52px",
          "heightInputNormal": "44px",
          "heightInputSmall": "32px",
          "heightRadioButton": "20px",
          "heightSeparator": "1px",
          "heightServiceLogoLarge": "48px",
          "heightServiceLogoMedium": "24px",
          "heightServiceLogoSmall": "12px",
          "lineHeightHeading": "1.2",
          "lineHeightText": "1.4",
          "marginBottomSelectSuffix": "2px",
          "marginButtonIconLarge": "12px",
          "marginButtonIconNormal": "8px",
          "marginButtonIconSmall": "8px",
          "marginRightBadgeIcon": "4px",
          "marginRightButtonGroup": "1px",
          "marginTopFormFeedback": "2px",
          "modifierScaleButtonActive": 0.95,
          "modifierScaleCheckboxRadioActive": 0.95,
          "opacityButtonDisabled": "0.3",
          "opacityCheckboxDisabled": "0.5",
          "opacityOverlay": "0.8",
          "opacityRadioButtonDisabled": "0.5",
          "paddingAlert": "16px",
          "paddingAlertWithIcon": "12px",
          "paddingButtonLarge": "28px",
          "paddingButtonLargeWithIcon": "16px",
          "paddingButtonNormal": "16px",
          "paddingButtonNormalWithIcon": "12px",
          "paddingButtonSmall": "12px",
          "paddingButtonSmallWithIcon": "8px",
          "paddingInputNormal": "0 12px",
          "paddingInputSmall": "0 12px",
          "paddingLeftInputFile": "6px",
          "paddingLeftSelectPrefix": "48px",
          "paddingTextareaNormal": "12px",
          "paddingTextareaSmall": "8px 12px",
          "paletteBlueDark": "#003A69",
          "paletteBlueLight": "#e0f6ff",
          "paletteBlueLightActive": "#b0e8fe",
          "paletteBlueLightHover": "#c8effe",
          "paletteBlueNormal": "#0176D2",
          "paletteBlueNormalActive": "#0064b2",
          "paletteBlueNormalHover": "#006abd",
          "paletteCloudLight": "#f5f7f9",
          "paletteCloudLightActive": "#d6dee6",
          "paletteCloudLightHover": "#e5eaef",
          "paletteCloudNormal": "#e8edf1",
          "paletteCloudNormalActive": "#cad5df",
          "paletteCloudNormalHover": "#d9e1e8",
          "paletteGreenDark": "#065d12",
          "paletteGreenLight": "#e7f3e8",
          "paletteGreenLightActive": "#c7e3c9",
          "paletteGreenLightHover": "#d7ebd8",
          "paletteGreenNormal": "#46B655",
          "paletteGreenNormalActive": "#3fa34c",
          "paletteGreenNormalHover": "#42ac50",
          "paletteInkDark": "#171b1e",
          "paletteInkLight": "#7f91a8",
          "paletteInkLightActive": "#5f738c",
          "paletteInkLightHover": "#6d819c",
          "paletteInkLighter": "#bac7d5",
          "paletteInkLighterActive": "#94a8be",
          "paletteInkLighterHover": "#a6b6c8",
          "paletteInkNormal": "#46515e",
          "paletteInkNormalActive": "#38404b",
          "paletteInkNormalHover": "#3f4854",
          "paletteOrangeDark": "#a93610",
          "paletteOrangeLight": "#fcf1cd",
          "paletteOrangeLightActive": "#f9e4a1",
          "paletteOrangeLightHover": "#faeab7",
          "paletteOrangeNormal": "#f9971e",
          "paletteOrangeNormalActive": "#e68206",
          "paletteOrangeNormalHover": "#f48a06",
          "paletteProductDark": "#00826f",
          "paletteProductLight": "#E1F5F2",
          "paletteProductLightActive": "#9addd3",
          "paletteProductLightHover": "#bde9e2",
          "paletteProductNormal": "#00a991",
          "paletteProductNormalActive": "#008f7b",
          "paletteProductNormalHover": "#009882",
          "paletteRedDark": "#650808",
          "paletteRedLight": "#fae8e8",
          "paletteRedLightActive": "#f1c0c0",
          "paletteRedLightHover": "#f5d4d4",
          "paletteRedNormal": "#d21c1c",
          "paletteRedNormalActive": "#b21717",
          "paletteRedNormalHover": "#bd1919",
          "paletteSocialFacebook": "#3b5998",
          "paletteSocialFacebookActive": "#354f88",
          "paletteSocialFacebookHover": "#385490",
          "paletteWhite": "#fff",
          "paletteWhiteActive": "#e5eaef",
          "paletteWhiteHover": "#f5f7f9",
          "spaceLarge": "24px",
          "spaceMedium": "16px",
          "spaceSmall": "12px",
          "spaceXLarge": "32px",
          "spaceXSmall": "8px",
          "spaceXXLarge": "40px",
          "spaceXXSmall": "4px",
          "spaceXXXLarge": "52px",
          "spaceXXXSmall": "2px",
          "textDecorationTextLinkPrimary": "none",
          "textDecorationTextLinkPrimaryHover": "underline",
          "textDecorationTextLinkSecondary": "underline",
          "textDecorationTextLinkSecondaryHover": "underline",
          "widthBadgeCircled": "24px",
          "widthCheckbox": "20px",
          "widthCountryFlag": "24px",
          "widthIconLarge": "32px",
          "widthIconMedium": "24px",
          "widthIconSmall": "16px",
          "widthRadioButton": "20px",
          "zIndexDefault": "1",
          "zIndexModal": "825",
          "zIndexModalOverlay": "800",
          "zIndexOnTheTop": "900",
          "zIndexSticky": "100",
        },
      }
    }
    viewBox="0 0 50 7"
  >
    <svg
      className="c0"
      data-test="test"
      preserveAspectRatio="xMidYMid meet"
      viewBox="0 0 50 7"
    >
      <Stops
        stops="4"
      >
        <path
          d="M36 3.53516c0 .82842.6716 1.5 1.5 1.5s1.5-.67158 1.5-1.5c0-.82843-.6716-1.5-1.5-1.5s-1.5.67157-1.5 1.5zm-.9965494.13248906C35.0011597 3.62377597 35 3.57960342 35 3.53516c0-1.38072 1.1193-2.5 2.5-2.5 1.3294839 0 2.416601 1.03778231 2.4954234 2.3474725-.0030675-.05274555-.007836-.10502861-.0142234-.1568025-.0081-.06421-.0183-.12769-.031-.19043h8.1709L45.7927.707275 46.5 0l3.5354 3.53564-3.5356 3.5354-.7073-.70703 2.3286-2.32861H39.95a2.4679841 2.4679841 0 0 0 .0008383-.00415845C39.720673 5.1743025 38.7108592 6.03516 37.5 6.03516c-1.21083 0-2.2206243-.86081602-2.4507698-2.00385522L35.05 4.0354h-4.1c.0017-.00854.0034-.01733.0051-.02588C30.7329 5.1626 29.718 6.03516 28.5 6.03516c-1.3188 0-2.3994-1.02149-2.4934-2.31641.0078.10767.0225.21338.0434.31665h-4.1c.0112-.05493.0205-.1106.028-.16675.0054-.03955.0098-.07934.0132-.11963.0056-.06714.0088-.13525.0088-.20385-.0007.16821-.0178.33154-.05.49023-.2322 1.14038-1.241 1.99976-2.45 1.99976-1.2087 0-2.2175-.85938-2.45-1.99976l-.007-.03589c.0022.01196.0046.02393.007.03589h-4.1c-.2317 1.14087-1.2408 1.99976-2.45 1.99976-1.20923 0-2.21826-.85889-2.44995-1.99976H0v-1h8.0498l-.00219.01123C8.2749 1.89966 9.28638 1.03516 10.5 1.03516c1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c.2314-1.14111 1.2407-2.00024 2.4502-2.00024 1.2095 0 2.2188.85913 2.4502 2.00024h4.0996c-.0034.01733-.0066.03467-.0098.052.211-1.16699 1.2322-2.05224 2.46-2.05224 1.3245 0 2.4087 1.031 2.4946 2.33374-.0014-.02369-.0034-.04737-.0056-.0708-.0085-.08912-.0215-.17676-.0388-.2627h4.0996c-.0327.16138-.0498.32861-.0498.49976 0 .04443.00115.0886175.0034375.13250125zm.0457796.36365572a2.4892309 2.4892309 0 0 1-.0118508-.06304804zm4.9015562-.00008263c.0168281-.08359295.0294864-.16869521.0377486-.25508048zM10.5 5.03516c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5c-.82837 0-1.5.67163-1.5 1.5s.67163 1.5 1.5 1.5zm7.5-1.5c0 .82837.6716 1.5 1.5 1.5s1.5-.67163 1.5-1.5-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5zm10.5 1.5c.8284 0 1.5-.67163 1.5-1.5s-.6716-1.5-1.5-1.5-1.5.67163-1.5 1.5.6716 1.5 1.5 1.5z"
        />
      </Stops>
    </svg>
  </StopoverArrow__StyledArrow>
</StopoverArrow>
`;
